### PR TITLE
[IMP] sale_order_line_position: allow to disable position recomputing

### PR DESCRIPTION
**Pull request by ajaniszewska-dev**
Add option which allows to disable line position recomputing

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]

File: sale_order_line_position/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2021 Camptocamp SA
```
Comment (grindtildeath) [inline sale_order_line_position/models/res_config_settings.py]:
```suggestion
# Copyright 2022 Camptocamp SA
```
Seems your code template are no configured properly :wink:

File: sale_order_line_position/models/res_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    disable_position_recompute = fields.Boolean(
+        related="company_id.disable_position_recompute",
+        readonly=False,
+        config_parameter="sale.disable_position_recompute",
```
Comment (grindtildeath) [inline sale_order_line_position/models/res_config_settings.py]: Why would you need a `ir.config_parameter` if the value is stored on the company? :thinking:

File: sale_order_line_position/models/res_company.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    disable_position_recompute = fields.Boolean(string="Do not recompute positions")
```
Comment (grindtildeath) [inline sale_order_line_position/models/res_company.py]:
```suggestion
    disable_sale_position_recompute = fields.Boolean(string="Do not recompute positions on sale orders")
```

File: sale_order_line_position/models/res_config_settings.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    disable_position_recompute = fields.Boolean(
```
Comment (grindtildeath) [inline sale_order_line_position/models/res_config_settings.py]:
```suggestion
    disable_sale_position_recompute = fields.Boolean(
```

File: sale_order_line_position/models/res_config_settings.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        related="company_id.disable_position_recompute",
```
Comment (grindtildeath) [inline sale_order_line_position/models/res_config_settings.py]:
```suggestion
        related="company_id.disable_sale_position_recompute",
```

File: sale_order_line_position/models/sale_order.py [hunk 6]
```diff
@@ -24,7 +24,9 @@ def action_quotation_send(self):
 
     def recompute_positions(self):
         for sale in self:
-            if sale.locked_positions:
+            if sale.locked_positions or self.env[
+                "ir.config_parameter"
+            ].sudo().get_param("sale.disable_position_recompute"):
```
Comment (grindtildeath) [inline sale_order_line_position/models/sale_order.py]:
```suggestion
            if sale.locked_positions or sales.company_id.disable_sale_position_recompute
```

File: sale_order_line_position/views/res_config_settings.xml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+                >
+                    <div
+                        class="col-12 col-lg-6 o_setting_box"
+                        id="line_positioning_options"
+                    >
+                        <div class="o_setting_left_pane">
+                            <field name="disable_position_recompute" />
+                        </div>
+                        <div class="o_setting_right_pane">
+                                <label for="disable_position_recompute" />
```
Comment (grindtildeath) [inline sale_order_line_position/views/res_config_settings.xml]:
```suggestion
                                <label for="disable_sale_position_recompute" />
```

File: sale_order_line_position/views/res_config_settings.xml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+                <div
+                    class="row mt16 o_settings_container"
+                    name="line_positioning_setting_container"
+                >
+                    <div
+                        class="col-12 col-lg-6 o_setting_box"
+                        id="line_positioning_options"
+                    >
+                        <div class="o_setting_left_pane">
+                            <field name="disable_position_recompute" />
```
Comment (grindtildeath) [inline sale_order_line_position/views/res_config_settings.xml]:
```suggestion
                            <field name="disable_sale_position_recompute" />
```

File: sale_order_line_position/views/res_config_settings.xml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="res_config_settings_view_form" model="ir.ui.view">
+        <field name="name">sale.config.positioning.form</field>
+        <field name="model">res.config.settings</field>
+        <field name="inherit_id" ref="base.res_config_settings_view_form" />
+        <field name="arch" type="xml">
+            <div id="connectors_setting_container" position="after">
+                <h2>Sale Order Line Positioning</h2>
```
Comment (grindtildeath) [inline sale_order_line_position/views/res_config_settings.xml]:
```suggestion
                <h2>Sale Order Line Positions</h2>
```

**Review by grindtildeath — CHANGES_REQUESTED** [review]
Some nitpicking, but mostly an important choice to do where you store the settings value.

**Review by ivantodorovich — CHANGES_REQUESTED** [review]
Agree with @grindtildeath's comments

**Review by ivantodorovich — APPROVED** [review]

**Review by TDu — APPROVED** [review]

**Review by cesar-tecnativa — APPROVED** [review]

**Comment by ajaniszewska-dev** [comment]
@pedrobaeza  can we merge that one?

**Pull request merged**